### ISSUE TITLE
PIN current workspace only

### ISSUE THREAD

**Issue by seriousgnu** [flameshot-org/flameshot#3488]
### Installation Type

Linux, MacOS, or Windows Package manager (apt, pacman, eopkg, choco, brew, ...), AppImage

### Operating System type and version

Ubuntu MATE 22.04

### PIN on workspace visibility

In v. 12.0+ screenshot pins are always visible on active/current workspace, where up until v. 11.0, pins would only be visible on the workspace (desktop) on which the screenshot was taken. Irrespective if Flameshot is installed via *deb, or installed using an AppImage file, the behavior is the same.

Related, Flameshot v.12 no longer shows a button in the window list (on the bottom/side/top panel).

I have reverted back to v. 11 because of this behavior (using the AppImage version), even though some v. 12 functionality is missing (e.g. the AppImage indicator icon is a generic icon, which may be an AppImage issue; the panel button in the window list cannot be hidden; the close/save/... context menu does not exist; etc.)

Suggestion: 1) option in configuration settings to allow panel button hidden or shown, and definitely 2) option to show screenshot pins on all workspaces or only the original workspace where the screenshot was taken.

### Steps to reproduce

1. Take screenshot on workspace 1.
2. Choose to pin screenshot.
3. Move to worksapce 2, 3, 4, etc.
4. Notice how screenshot taken on workspace 1 is also visible on workspace 2, 3, 4, etc. (which should not be a default behavior; it could/should at the very least be optional to prevent this always-visible behavior).

### System Information

Ubuntu MATE 22.04, Xorg (X11).

**Comment by mmahmoudian** [comment]
> In v. 12.0+ screenshot pins are always visible on active/current workspace

Correct. This is the intended behavior. I personally liked it to have the previous functionality but [I believe] because it does not have it's own panel item, it was decided to be always visible to user.

> Related, Flameshot v.12 no longer shows a button in the window list (on the bottom/side/top panel).

Would you please explain this more. I don't understand what button your are referring to. If possible please use v12 and v11 AppImages and take a screenshot of the differences.

> AppImage indicator icon is a generic icon, which may be an AppImage issue;

For AppImage icon, please refer to: https://github.com/AppImage/AppImageKit/issues/1137

> Suggestion: 1) option in configuration settings to allow panel button hidden or shown, and definitely 2) option to show screenshot pins on all workspaces or only the original workspace where the screenshot was taken.

Then I will mark this as feature request and remove bug report.

**Comment by seriousgnu** [comment]
> > In v. 12.0+ screenshot pins are always visible on active/current workspace
> 
> Correct. This is the intended behavior. I personally liked it to have the previous functionality but [I believe] because it does not have it's own panel item, it was decided to be always visible to user.
 
Did not realize this was intended. So, sure, bug, request, ... :) I do find this one-screenshot-one-workspace item important though. When, for instance, I take a screenshot of an item on the workspace where I am paying my bills -- to be reminded later of, say, the amount or account number of the recipient -- I do not want that same screenshot mixing in with other screenshots I am taking on the workspace where I am programming the next big AI app. (I wish I was, but you get the drift, I am sure.)

> > Related, Flameshot v.12 no longer shows a button in the window list (on the bottom/side/top panel).
> 
> Would you please explain this more. I don't understand what button your are referring to. If possible please use v12 and v11 AppImages and take a screenshot of the differences.

I am attaching a screenshot of a screenshot I took of part of my desktop wallpaper. With v. 11, you get to see the 'Flameshot' window button in the 'Window List' on the bottom panel. I included the context menu from where I could send the screenshot to any other workspace on my system. (Choosing 'Always group windows' in the preferences keeps the clutter to a minimum, and still allows me to select which screenshot I individually want to move to another workspace if needed.)

Note that to include the context menu, I had to install v. 11 from *deb since I could not immediately get to assign a keyboard shortcut that did more than just start another instance of Flameshot. (Assigning '/usr/bin/flameshot gui --delay 2000' to the PrtScrn key does the job, plus the *deb version gives back the nice Flameshot icon in the indicator.)

> > AppImage indicator icon is a generic icon, which may be an AppImage issue;
> 
> For AppImage icon, please refer to: [AppImage/AppImageKit#1137](https://github.com/AppImage/AppImageKit/issues/1137)
> 
> > Suggestion: 1) option in configuration settings to allow panel button hidden or shown, and definitely 2) option to show screenshot pins on all workspaces or only the original workspace where the screenshot was taken.
> 
> Then I will mark this as feature request and remove bug report.

Thanks a bunch.

![IMG-window-list-preferences](https://github.com/flameshot-org/flameshot/assets/159030722/b4ad64be-edf7-41dd-b96d-3b534c9f8fd1)
![IMG-flameshot-panel-button](https://github.com/flameshot-org/flameshot/assets/159030722/ab903c60-68ff-48b0-8d65-87b1cc452dbe)
